pkg/querier/worker/scheduler_processor.go

```diff
@@ -222,6 +226,15 @@ func (sp *schedulerProcessor) runRequest(ctx context.Context, logger log.Logger,
 			HttpResponse: response,
 			Stats:        stats,
 		})
+		// If the used connection returned and error, remove it from the pool and retry.
+		if err != nil && retries < maxNotifyFrontendRetries {
+			level.Warn(logger).Log("msg", "retrying to notify frontend about finished query", "err", err, "frontend", frontendAddress)
```

Comment:
Could the log message include the number of the retry?

Reply:
changed @ 51512047b4d488e3d7aa25e20c2fad8fd10d82ab